tests/unit/check_cl_cdm.c

```diff
@@ -0,0 +1,236 @@
+ */
+void teardown(void) { mock_hostmod_teardown(); }
+
+
+START_TEST(test_get_desc)
+{
+    osd_result rv;
+
+    mock_hostmod_expect_mod_describe(cdm_diaddr, OSD_MODULE_VENDOR_OSD,
+                                     OSD_MODULE_TYPE_STD_CDM, 0);
+
+    mock_hostmod_expect_reg_read16(1, cdm_diaddr, OSD_REG_CDM_CORE_CTRL,
+                                   OSD_OK);
+
+    mock_hostmod_expect_reg_read16(5, cdm_diaddr, OSD_REG_CDM_CORE_REG_UPPER,
+                                   OSD_OK);
+
+    mock_hostmod_expect_reg_read16(32, cdm_diaddr, OSD_REG_CDM_CORE_DATA_WIDTH,
+                                   OSD_OK);
+    
+    struct osd_cdm_desc cdm_desc;
+    rv = osd_cl_cdm_get_desc(mock_hostmod_get_ctx(), cdm_diaddr, &cdm_desc);
+    ck_assert_int_eq(rv, OSD_OK);
+    ck_assert_uint_eq(cdm_desc.core_ctrl, 1);
+    ck_assert_uint_eq(cdm_desc.core_reg_upper, 5);
+    ck_assert_uint_eq(cdm_desc.core_data_width, 32);
+    ck_assert_uint_eq(cdm_desc.di_addr, cdm_diaddr);
+}
+END_TEST
+
+START_TEST(test_get_desc_wrong_module)
+{
+    osd_result rv;
+
+    mock_hostmod_expect_mod_describe(cdm_diaddr, OSD_MODULE_VENDOR_OSD,
+                                     OSD_MODULE_TYPE_STD_CTM, 0);
+
+    struct osd_cdm_desc cdm_desc;
+    rv = osd_cl_cdm_get_desc(mock_hostmod_get_ctx(), cdm_diaddr, &cdm_desc);
+    ck_assert_int_eq(rv, OSD_ERROR_WRONG_MODULE);
+}
+END_TEST
+
+static void event_handler(void *arg, const struct osd_cdm_desc *cdm_desc,
+                          const struct osd_cdm_event *event)
+{
+    ck_assert(arg);
+    ck_assert(cdm_desc);
+    ck_assert(event);
+
+    // in this test we pass the expected event as callback argument
+    struct osd_cdm_event *exp_event = arg;
+    ck_assert_uint_eq(exp_event->stall, event->stall);
+    
+}
+
+#define STALL_BIT 0
+
+START_TEST(test_handle_event)
+{
+    osd_result rv;
+
+    struct osd_cdm_desc cdm_desc;
+    cdm_desc.di_addr = 2;
+    cdm_desc.core_ctrl = 16;
+    cdm_desc.core_reg_upper = 1;
+    cdm_desc.core_data_width = 32;
+
+    struct osd_cdm_event_handler ev_handler;
+    ev_handler.cb_fn = event_handler;
+    ev_handler.cdm_desc = &cdm_desc;
+
+    struct osd_packet *pkg_trace;
+    osd_packet_new(&pkg_trace, osd_packet_sizeconv_payload2data(1));
+    rv = osd_packet_set_header(pkg_trace, 1, 2, OSD_PACKET_TYPE_EVENT, 0);
+    ck_assert_int_eq(rv, OSD_OK);
+    pkg_trace->data.payload[0] = BIT(STALL_BIT); // stall (1 << 0)
+
+    struct osd_cdm_event exp_event;
+    exp_event.stall = BIT(STALL_BIT);
+    ev_handler.cb_arg = (void*)&exp_event;
+    rv = osd_cl_cdm_handle_event((void*)&ev_handler, pkg_trace);
+    ck_assert_int_eq(rv, OSD_OK);
+}
+END_TEST
+
+struct osd_cdm_desc get_cdm_desc(void)
+{
+    struct osd_cdm_desc cdm_desc = { 0 };
+    cdm_desc.di_addr = cdm_diaddr;
+    cdm_desc.core_ctrl = 1;
+    cdm_desc.core_reg_upper = 0;
+    cdm_desc.core_data_width = 32;
+
+    return cdm_desc;
+}
+
+START_TEST(test_cpu_read_register_test1)
```

Comment:
Please use a more descriptive name than "test1". If you cannot use a descriptive function name add a doxygen comment to the function. (Apply this to all four test functions.)